host/sgx/create.c

```diff
@@ -589,57 +624,85 @@ oe_result_t oe_sgx_validate_enclave_properties(
 }
 
 #ifdef OE_WITH_EXPERIMENTAL_EEID
-static oe_result_t _add_eeid_marker_page(
+static oe_result_t _save_eeid_info(
     oe_sgx_load_context_t* context,
-    oe_enclave_t* enclave,
-    size_t image_size,
     uint64_t entry_point,
     oe_sgx_enclave_properties_t* props,
     uint64_t* vaddr)
 {
-    oe_result_t result = OE_UNEXPECTED;
-    oe_eeid_t* eeid = context->eeid;
+    const oe_enclave_setting_eeid_t* eeid_setting = context->eeid_setting;
 
-    if (eeid && is_eeid_base_image(props) &&
-        context->type == OE_SGX_LOAD_TYPE_CREATE)
+    if (eeid_setting)
     {
-        // Finalize the memory settings
-        props->header.size_settings = eeid->size_settings;
+        /* EEID must fit on a single page. */
+        size_t signature_size = sizeof(sgx_sigstruct_t);
+        size_t eeid_size = sizeof(oe_eeid_t) + signature_size;
+        if (eeid_size > OE_PAGE_SIZE)
+            return OE_INVALID_PARAMETER;
+
+        oe_eeid_t* eeid = NULL;
+        oe_create_eeid_sgx(&eeid);
+
+        /* Finalize the memory settings */
+        if (context->type == OE_SGX_LOAD_TYPE_CREATE &&
+            is_eeid_base_image(&props->header.size_settings))
+            props->header.size_settings = eeid_setting->size_settings;
+        else
+            eeid->size_settings = props->header.size_settings;
 
-        // Record EEID information
+        /* Record EEID information */
         eeid->version = OE_EEID_VERSION;
         oe_sha256_context_t* hctx = &context->hash_context;
         oe_sha256_save(hctx, eeid->hash_state.H, eeid->hash_state.N);
+        eeid->size_settings = props->header.size_settings;
         eeid->entry_point = entry_point;
         eeid->vaddr = *vaddr;
-        eeid->signature_size = sizeof(sgx_sigstruct_t);
-        memcpy(
-            eeid->data + eeid->data_size,
-            (uint8_t*)&props->sigstruct,
-            sizeof(sgx_sigstruct_t));
+        eeid->signature_size = signature_size;
+        memcpy(eeid->signature, props->sigstruct, signature_size);
+
+        /* Compute config hash (all zero during signing) */
```

Comment:
During the base image signing, EEID page should not be measured. Does the comment and implementaiton here mean EEID page with `config_id`=0 is part of the base image measurement?

Reply:
No, during signing this function is only used to record relevant information (like the hash state) so that `oesign dump-eeid` can dump it, but nothing is being measured. The signing case is here: https://github.com/openenclave/openenclave/blob/204830a5721b796684c62268474d8ce66ddc9e08/host/sgx/create.c#L769 and it doesn't call `_add_eeid_page` or `_eeid_resign`.